pkg/controller/scyllacluster/sync_certs.go

```diff
@@ -236,14 +293,23 @@ func (scc *Controller) syncCerts(
 	}
 
 	// For every cluster domain we'll create "cql" subdomain and "UUID.cql" subdomains for every node.
-	var servingDNSNames []string
 	for _, domain := range sc.Spec.DNSDomains {
 		servingDNSNames = append(servingDNSNames, naming.GetCQLProtocolSubDomain(domain))
 
 		for _, hostID := range hostIDs {
 			servingDNSNames = append(servingDNSNames, naming.GetCQLHostIDSubDomain(hostID, domain))
 		}
 	}
+
+	// Sign for every node DNS name and discovery endpoint.
```

Comment:
I think I don't understand why it's necessary to introduce it in this PR? Could you please explain?

Reply:
When node service type is Headless, we end up with chicken egg problem for serving certificate. Certificate cannot have empty SANs, and we need to sign it for PodIP. Pod IP is allocated after Pod initialization, for which we need all non-optional volumes to be available. Serving cert is non-optional, and it should stay like that. 

To solve it, we sign certificate with in-cluster DNS names which is what is expected to be in the cert, but was missing.

Reply:
I see, thanks!